workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -654,118 +654,142 @@ for (var i = 0; i < commandDetailsList.Count; i++) {
-                ComponentType.ReadOnly<ComponentRemoved<<#= componentNamespace #>.Component>>(),
-            };
-
-            public override ComponentType[] EventComponentTypes => new ComponentType[] {
+            public override EntityArchetypeQuery CleanupArchetypeQuery => new EntityArchetypeQuery
+            {
+                All = Array.Empty<ComponentType>(),
+                Any = new ComponentType[]
+                {
+                    ComponentType.ReadOnly<ComponentAdded<<#= componentNamespace #>.Component>>(),
+                    ComponentType.ReadOnly<ComponentRemoved<<#= componentNamespace #>.Component>>(),
+                    ComponentType.ReadOnly<<#= componentNamespace #>.ReceivedUpdates>(),
+                    ComponentType.ReadOnly<AuthorityChanges<<#= componentNamespace #>.Component>>(),
 <# foreach (var eventDetails in eventDetailsList) { #>
-                ComponentType.ReadOnly<ReceivedEvents.<#= eventDetails.EventName #>>(),
+                    ComponentType.ReadOnly<ReceivedEvents.<#= eventDetails.EventName #>>(),
 <# } #>
-            };
-
-            public override ComponentType ComponentUpdateType => ComponentType.ReadOnly<<#= componentNamespace #>.ReceivedUpdates>();
-            public override ComponentType AuthorityChangesType => ComponentType.ReadOnly<AuthorityChanges<<#= componentNamespace #>.Component>>();
-
-            public override ComponentType[] CommandReactiveTypes => new ComponentType[] {
 <# foreach (var commandDetails in commandDetailsList) { #>
-                ComponentType.ReadOnly<CommandRequests.<#= commandDetails.CommandName #>>(),
-                ComponentType.ReadOnly<CommandResponses.<#= commandDetails.CommandName #>>(),
+                    ComponentType.ReadOnly<CommandRequests.<#= commandDetails.CommandName #>>(),
+                    ComponentType.ReadOnly<CommandResponses.<#= commandDetails.CommandName #>>(),
 <# } #>
+                },
+                None = Array.Empty<ComponentType>(),
             };
 
-            public override void CleanupUpdates(ComponentGroup updateGroup, ref EntityCommandBuffer buffer)
+            public override void CleanComponents(ComponentGroup group, ComponentSystemBase system,
+                EntityCommandBuffer buffer)
             {
-                if (updateGroup.IsEmptyIgnoreFilter)
-                {
-                    return;
-                }
+                var entityType = system.GetArchetypeChunkEntityType();
+                var componentAddedType = system.GetArchetypeChunkComponentType<ComponentAdded<<#= componentNamespace #>.Component>>();
+                var componentRemovedType = system.GetArchetypeChunkComponentType<ComponentRemoved<<#= componentNamespace #>.Component>>();
+                var receivedUpdateType = system.GetArchetypeChunkComponentType<<#= componentNamespace #>.ReceivedUpdates>();
+<# for (var i = 0; i < eventDetailsList.Count; i++) { #>
+<# var eventDetails = eventDetailsList[i]; #>
```

Comment:
Nit: You can combine these statements in the same `<# #>` block like with commands

Reply:
Will change